src/pages/ThorChainLP/Pool/components/PoolChart.tsx

```diff
@@ -1,34 +1,130 @@
 import { Button, ButtonGroup, Center, Flex, Stack } from '@chakra-ui/react'
-import { FaChartSimple } from 'react-icons/fa6'
-import { useTranslate } from 'react-polyglot'
-import { RawText } from 'components/Text'
+import { thorchainAssetId } from '@shapeshiftoss/caip'
+import { useQuery } from '@tanstack/react-query'
+import type { SingleValueData } from 'lightweight-charts'
+import { useCallback, useMemo, useState } from 'react'
+import { reactQueries } from 'react-queries'
+import type { Interval } from 'react-queries/queries/midgard'
+import { SimpleChart } from 'components/SimpleChart'
+import { fromThorBaseUnit } from 'lib/utils/thorchain'
+import type {
+  MidgardSwapHistoryResponse,
+  MidgardTvlHistoryResponse,
+} from 'lib/utils/thorchain/lp/types'
+import { selectMarketDataByAssetIdUserCurrency } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+const swapHistoryToChartData = (
+  swapHistory: MidgardSwapHistoryResponse | undefined,
+  runePrice: string,
+): SingleValueData<number>[] => {
+  if (!swapHistory) return []
+
+  return swapHistory.intervals.map(interval => {
+    const intervalVolume2FiatUserCurrency = fromThorBaseUnit(interval.totalVolume)
+      .times(runePrice)
+      .toFixed()
+
+    return {
+      time: Number(interval.startTime),
+      value: Number(intervalVolume2FiatUserCurrency),
+    }
+  })
+}
+
+const tvlToChartData = (
+  tvl: MidgardTvlHistoryResponse,
+  runePrice: string,
+  thorchainNotationAssetId: string,
+): SingleValueData<number>[] =>
+  tvl.intervals.map(interval => {
+    const poolDepth = interval.poolsDepth.find(pool => pool.pool === thorchainNotationAssetId)
+    const poolTotalDepth = poolDepth?.totalDepth ?? '0'
+
+    const tvlFiat = fromThorBaseUnit(poolTotalDepth).times(runePrice).toFixed()
+    return {
+      time: Number(interval.startTime),
```

Comment:
the docs seem to imply this usage if we want to use the default types and not specify the generic. doesn't save us much though lol
`Number(interval.startTime) as UTCTimestamp`

Reply:
e67dd131dc9dab316cd52276d83f9c06025f7383